src/_guides/language/effective-dart/usage.md

```diff
@@ -245,87 +245,65 @@ void error([String? message = null]) {
 {% endprettify %}
 
-Both operations produce the same result and do the right thing, but `??` is
-preferred for three main reasons:
-
-*   The `??` operator signals that the code has something to do with `null`.
-
-*   The `== true` looks like a common mistake where the equality operator is
-    redundant and can be removed. That's true when the boolean expression on the
-    left will not produce `null`, but not when it can.
-
-*   The `?? false` and `?? true` clearly show what value will be used when the
-    expression is `null`. With `== true`, you have to think through the boolean
-    logic to realize that means that a `null` gets converted to *false*.
-
-**Exception:** Using a null-aware operator on a variable inside a condition
-doesn't promote the variable to a non-nullable type. If you want the variable
-to be promoted inside the body of the `if` statement, it might be better to use
-an explicit `!= null` check instead of `?.` followed by `??`:
+To evaluate a Boolean expression that *is nullable*, you should only use `??`
+or an explicit `!= null` check.
 
 {:.good}
-<?code-excerpt "usage_good.dart (null-aware-promote)"?>
 {% prettify dart tag=pre+code %}
-int measureMessage(String? message) {
-  if (message != null && message.isNotEmpty) {
-    // message is promoted to String.
-    return message.length;
-  }
+// If you want null to result in false:
+if (nullableBool ?? false) { ... }
 
-  return 0;
-}
+// If you want null to result in false
+// and you want the variable to type promote:
+if (nullableBool != null && nullableBool) { ... }
 {% endprettify %}
 
 {:.bad}
-<?code-excerpt "usage_bad.dart (null-aware-promote)"?>
 {% prettify dart tag=pre+code %}
-int measureMessage(String? message) {
-  if (message?.isNotEmpty ?? false) {
-    // message is not promoted to String.
-    return message!.length;
-  }
+// Error if null 
+if (nullableBool) { ... }
 
-  return 0;
-}
+// If you want null to be false:
+if (nullableBool == true) { ... }
 {% endprettify %}
 
+`if (nullableBool == true)` is a viable expression, but shouldn't be used for several reasons:
+
+* It doesn't indicate the code has anything to do with `null`.
+
+* Because it's not evidently `null` related, it can easily be mistaken for the non-nullable case,
+where the equality operator is redundant and can be removed.
+That’s only true when the boolean expression on the left has no chance of producing null, but not when it can.
+
+* The boolean logic is confusing. If `nullableBool` is null, then `nullableBool == true`
+means the condition evaluates as false.
+
+The `??` operator makes it clear that something to do with null is happening,
+so it won't be mistaken for a redundant operation. The logic is much clearer too; 
+the result of the expression being null is the same as the Boolean literal.
+This is commonly considered "converting" `null` to `true` or `false` to be able to
+pass the result of the evaluation to something that expects a non-nullable Boolean value.
```

Comment:
I'd probably just remove the sentence on "converting". I don't think it adds much value. That was how I framed it originally, but I don't know if that's how anyone actually thinks about it.

Reply:
@munificent The reason I kept that point from the original section is because of the lint `use_if_null_to_convert_nulls_to_bools` that currently points to the section I rewrote/removed. 

I redirected that lint to point to this new section, but the new content doesn't mention anything about "converting" so it's...odd. 

I will raise this with the analyzer team (and @ you there) to change the lint if no one thinks about  it like "converting" anyway.

Reply:
Ok based on https://github.com/dart-lang/linter/issues/3908 it seems safe to remove that part, thanks!